main.rb

```diff
@@ -0,0 +1,95 @@
+module Enumerable
+
+    def my_each
+    unless !block_given?
+      for item in self
+        yield(item)
+      end
+    end
+      return self
```

Comment:
When it runs out of code or "reaches the end", in which case the value of the last expression is returned. So you don't need to type `return` in here. Just `self` will do it.